### ISSUE TITLE
[BUG] rules#call-non-callable

### ISSUE THREAD

**Issue by LIghtJUNction** [astral-sh/ty#3412]
### Summary

Here is an example. When the judgment logic becomes complex and I don't want to run the judgment logic repeatedly, I might use a boolean to store the judgment result, but this will cause the type checker to report an error. Here is an example.

```python
a: A | None = A()
a_is_none: bool = a is None

if a_is_none:
    a() # Object of type `None` is not callablety[call-non-callable](https://ty.dev/rules#call-non-callable)
((...) -> Unknown) | None

if a is None:
    a() # ok

```
I'm using the TY plugin for VS Code, version 2026.44.0.

### Version

2026.44.0

**Comment by MichaReiser** [comment]
I'm not sure I fully understand your example, but this should work with ty 0.0.33 or newer

`main.py`:

```py
class A: 
    
    def __call__(self, *args): ...

a: A | None = A()
a_is_not_none: bool = a is not None

if a_is_not_none:
    a()

if a is not None:
    a() # ok

```

[Playground](https://play.ty.dev/4ad92ec8-b6f9-4507-bf2c-e06dbd59ccc6)
